src/org/mozartoz/truffle/translator/Translator.java

```diff
@@ -272,7 +277,9 @@ OzNode translate(StatOrExpr node) {
 			}
 			return SequenceNode.sequence(decls.toArray(new OzNode[decls.size()]), translate(local.body()));
 		} else if (node instanceof CallCommon) {
-			return translateCall((CallCommon) node);
+			return new TailCallCatcherNode(translateCall((CallCommon) node));
```

Comment:
I think this should move in `translateCall`, because all call nodes need to have the catcher.